core/main/src/firebolt/handlers/metrics_rpc.rs

```diff
@@ -508,6 +510,35 @@ impl MetricsServer for MetricsImpl {
         media_progress_params: MediaProgressParams,
     ) -> RpcResult<bool> {
         let progress = convert_to_media_position_type(media_progress_params.progress)?;
+
+        if self
+            .state
+            .get_device_manifest()
+            .configuration
+            .default_values
+            .media_progress_as_watched_events
+            && progress != MediaPositionType::None
+        {
+            let request = WatchedRequest {
+                context: ctx.clone(),
+                info: WatchedInfo {
+                    entity_id: media_progress_params.entity_id.clone(),
+                    progress: media_progress_params
+                        .progress
+                        .expect("progress shall not be None at this point"),
```

Comment:
I feel like this should be written so `expect` isn't used here, it's similar to doing:

```
if some_option.is_none() {
  return;
}

do_something(some_option.unwrap());
```

...which we've since stopped doing in our code.

Also, could [watched()](https://github.com/rdkcentral/Ripple/blob/ad1f9c30581178635a820e4724a412a182dcaa59/core/main/src/firebolt/handlers/discovery_rpc.rs#L468) just be called instead of duplicating the code, we could also log an error if the request fails.

Reply:
Understood but my point was `expect` is much like `unwrap` which we've been cleaning up.

I think we call other rpc handlers outside of the associated module for similar reasons of code duplication. Could you at least add a log if the call fails?

Reply:
As suggested, removed the usage of `expect` . Added error log messages on failure to send watched event.